lib/setup/setupInstall.js

```diff
@@ -781,20 +781,21 @@ function Install(options) {
             });
     };
 
-    function callInstallOfAdapter(adapter, config, callback) {
-        const path_ = tools.getAdapterDir(adapter);
-
-        if (config.common.install && fs.existsSync(path_ + '/io-package.json')) {
+    async function callInstallOfAdapter(adapter, config, callback) {
+        if (config.common.install) {
             // Install node modules
-            const exec = require('child_process').exec;
+            const { exec } = require('child_process');
             let cmd = 'node ';
 
-            let fileName = config.common.main || 'main.js';
-            if (!fs.existsSync(path_ + '/' + fileName)) {
-                fileName = adapter + '.js';
+            let fileFullName;
+            try {
+                fileFullName = await tools.getAdapterMainFile(adapter);
+            } catch (e) {
+                return void callback(adapter);
             }
-            cmd += '"' + path  + '/' + fileName + '" --install';
```

Comment:
This could not work, because it had to be `path_` instead od path, because `path` represents the module

Reply:
TypeScript to the rescue :see_no_evil: